src/components/menu/menu.tsx

```diff
@@ -67,6 +105,37 @@ export class Menu {
     @Prop({ reflect: true })
     public gridLayout = false;
 
+    /**
+     * A search function that takes a search-string as an argument,
+     * and returns a promise that will eventually be resolved with
+     * an array of `MenuItem`:s.
+     *
+     * See the docs for the type `Searcher` for type information on
+     * the searcher function itself.
+     */
+    @Prop()
+    public searcher: MenuSearcher;
+
+    /**
+     * Making it possible for a consumer to tell the component that it's waiting for something.
+     * For example loading items/root items.
+     */
+    @Prop()
+    public loading: boolean;
+
+    /**
+     * Message to display when search returns 0 results.
+     */
+    @Prop()
+    public emptyResultMessage?: string;
+
+    /**
+     * Making it possible for the consumer so explicitly tell the component
+     * what has been clicked in the menu.
+     */
+    @Prop({ mutable: true })
+    public clickedMenuItem: MenuItem;
```

Comment:
I don't think we should use words such as `click` in the props, since you can navigate the menu with the keyboard as well. Perhaps `selectedMenuItem`?

Reply:
Good point, fixed this as well.